### PR TITLE
[readspeed] Misc improvements and fixes

### PR DESCRIPTION
**Pull request by eguiraud**
This PR is better reviewed commit by commit.

### CODE REVIEW

**Comment by eguiraud** [comment]
Rebased on top of master to remove the noimt fix that has been factored out to #11463 .

**Comment by eguiraud** [comment]
Rebased to fix conflicts.

File: tree/readspeed/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+## Interpreting results:
+
+### There are three possible scenarios when using rootreadspeed, namely:
+
+  - The 'Real Time' is significantly lower than your own analysis runtime.
+    This would imply your actual application code is dominating the runtime of your analysis,
+    ie. your analysis logic is taking up the time.
+    The best way to decrease the runtime would be to optimize your code, attempt to parallelize
+    it onto multiple threads if possible, or use a machine with a more performant CPU.
```
Comment (Axel-Naumann) [inline tree/readspeed/README.md:29]: Maybe add "Consider using RDataFrame with `ROOT::EnableImplicitMT()` and / or sharing your issue on https://root-forum.cern.ch"?

File: tree/readspeed/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    * If no '--threads' argument was provided this is 1, otherwise it is the minimum of the value
+      provided and the number of threads your CPU can run in parallel. It is worth noting that -
+      on shared systems or if running other heavy applications - the number of your own threads
+      running at any time may be lower than the limit due to demand on the CPU.
+  - The 'Real Time' is similar to 'CPU Time / number of threads' AND 'Compressed Throughput' is lower than expected
+    for your storage medium: this would imply that your CPU threads aren't decompressing data as fast as your storage
+    medium can provide it, and so decompression is the bottleneck.
+    The best way to decrease your runtime would be to utilise a system with a faster CPU, or make use
+    use of more threads when running, or use a compression algorithm with a higher decompression rate,
+    possibly at the cost of some extra file size.
```
Comment (Axel-Naumann) [inline tree/readspeed/README.md:49]: "such as LZ4"?

File: tree/readspeed/src/ReadSpeedCLI.cxx [hunk 3]
```diff
@@ -129,7 +129,18 @@ const auto fullUsageText =
...
-   "  (On Linux this can be done by running 'echo 3 > /proc/sys/vm/drop_caches' as a superuser)\n";
+   " On Linux this can be done by running 'echo 3 > /proc/sys/vm/drop_caches' as a superuser\n"
+   " or a specific file can be dropped from the cache with"
+   " `dd of=<FILENAME> oflag=nocache conv=notrunc,fdatasync count=0 > /dev/null 2>&1`.\n"
+   "\n"
+   "\n"
+   "Known overhead of TTreeReader, RDataFrame:\n"
+   "`rootreadspeed` is designed to read all data present in the specified branches, trees and files at the highest\n"
+   "possible speed. When an analysis application is bound by I/O, higher-level interfaces built on top of TTree\n"
+   "such as TTreeReader and RDataFrame are known to add a significant runtime overhead with respect to the runtimes\n"
```
Comment (Axel-Naumann) [inline tree/readspeed/src/ReadSpeedCLI.cxx:140]: If the analysis is I/O bound then extra CPU cycles shouldn't matter? Not sure I understand this...

File: tree/readspeed/src/ReadSpeedCLI.cxx [hunk 4]
```diff
@@ -129,7 +129,18 @@ const auto fullUsageText =
...
+   "reported by `rootreadspeed` (up to a factor 2). In realistic analysis applications it has been observed that a\n"
```
Comment (Axel-Naumann) [inline tree/readspeed/src/ReadSpeedCLI.cxx:141]: Can you add that this is planned to be addressed by `RDataFrame`'s usage of `RNTuple`?

File: tree/readspeed/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+If your data is stored on a local disk, the system may cache some/all of the file in memory after it is
+first read. If this is realistic of how your analysis will run - then there is no concern. However, if
+you expect to only read files once in a while - and as such the files are unlikely to be in the cache -
+consider clearing the cache before running rootreadspeed.
+On Linux this can be done by running 'echo 3 > /proc/sys/vm/drop_caches' as a superuser,
+or a specific file can be dropped from the cache with
+`dd of=<FILENAME> oflag=nocache conv=notrunc,fdatasync count=0 > /dev/null 2>&1`.
+
+
+### Known overhead of TTreeReader, RDataFrame
```
Comment (Axel-Naumann) [inline tree/readspeed/README.md:63]: See my comments below, in the help for the cli tool.

**Review by Axel-Naumann — APPROVED** [review]
But see comments.

File: tree/readspeed/src/ReadSpeedCLI.cxx (hunk 3 above)
Comment (eguiraud) [reply tree/readspeed/src/ReadSpeedCLI.cxx:140]: Here it's I/O as in "the ROOT I/O machinery" (which `rootreadspeed` can use more efficiently than `TTreeReader` does).

I'll change it to "When the bottleneck is not in the analysis logic..."

**Pull request merged**